back/tools/install-mariadb

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+set -e
+
+OSLIN="linux"
+OSDAR="darwin"
+ostype="$([[ "${OSTYPE}" == ${OSDAR}* ]] && echo ${OSDAR} || echo ${OSLIN})"
+OSWSL="linux-wsl"
+[[ "${ostype}" == ${OSLIN} ]] && grep -qi "Microsoft\|WSL" /proc/version && ostype=${OSWSL}
+
+
+if ! hash mysql 2>/dev/null; then
+	case ${ostype} in
+	${OSLIN}*)
+		sudo apt update
+		sudo apt install software-properties-common
+
+		pkey="0xF1656F24C74CD1D8"
+		case ${ostype} in
+		${OSLIN})
+			keysrv="hkp://keyserver.ubuntu.com:80"
+			sudo apt-key adv --recv-keys --keyserver "${keysrv}" "${pkey}"
+			;;
+		${OSWSL})
+			keysrch="http://keyserver.ubuntu.com/pks/lookup?op=get&search=${pkey}"
+			curl -sL "${keysrch}" | sudo apt-key add
+			;;
+		esac
+		repourl=http://sfo1.mirrors.digitalocean.com/mariadb/repo/10.3/ubuntu 
+		sudo add-apt-repository "deb [arch=amd64] ${repourl} $(lsb_release -cs) main"
+		sudo apt update
+		sudo apt install mariadb-server
+		;;
+	${OSDAR})
+		brew update
+		brew install mariadb
+		;;
+	esac
+
+	sudo cat << EOF > /etc/mysql/mariadb.conf.d/60-encoding.cnf
```

Comment:
This line doesn't run with sudo in wsl, and gives a `permission denied` error. But if you `sudo ./install-mariadb`, then it plows through till the end...

Reply:
This should work properly now.

The incorrect code is using a "heredoc" to send data to `cat` that is ran using `sudo`, but it is opening the file descriptor "/etc/mysql/mariadb.conf.d/60-encoding.cnf" without `sudo`. By using `tee`, the intended destination descriptor can be opened with `sudo` and the data from the heredoc is then stored properly. Because of how `tee` works, we must discard some data to "/dev/null".